### PR TITLE
New Component: Time Slider

### PR DESCRIPTION
**Pull request by njarecha**

### CODE REVIEW

**Comment by mhdostal** [comment]
@njarecha - this has all been reviewed already, correct?

**Comment by njarecha** [comment]
@mhdostal Yes

File: Toolkit/ArcGISToolkit/TimeSlider.swift [hunk 1]
```diff
@@ -0,0 +1,2727 @@
...
+    private let paddingBetweenLabels: CGFloat = 5.0
+    
+    private var tickMarkLabels = [CATextLayer]()
+    fileprivate var tickMarks = [TickMark]()
+    
+    private var playPauseButton = UIButton(type: .custom)
+    private var forwardButton = UIButton(type: .custom)
+    private var backButton = UIButton(type: .custom)
+    
+    private var playImage: UIImage?
```
Comment (philium) [inline Toolkit/ArcGISToolkit/TimeSlider.swift]: These image properties are only being used once and only inside of `setupUI()`. Could we get rid of them by either:
1. Moving them into `setupUI()`, or
1. Creating and setting the images simultaneously: `setImage(UIImage(named: …`

File: Toolkit/ArcGISToolkit/TimeSlider.swift [hunk 2]
```diff
@@ -0,0 +1,2727 @@
...
+    private let maximumThumbSize: CGFloat = 50.0
+    private var trackLayerSidePadding: CGFloat = 30.0
+    private let labelSidePadding: CGFloat = 10.0
+    private let labelPadding: CGFloat = 3.0
+    private let paddingBetweenLabels: CGFloat = 5.0
+    
+    private var tickMarkLabels = [CATextLayer]()
+    fileprivate var tickMarks = [TickMark]()
+    
+    private var playPauseButton = UIButton(type: .custom)
```
Comment (philium) [inline Toolkit/ArcGISToolkit/TimeSlider.swift]: These button properties are never mutated after initialization. Could you make them constants (`let`)?

File: Toolkit/ArcGISToolkit/TimeSlider.swift [hunk 3]
```diff
@@ -0,0 +1,2727 @@
...
+        
+        // Setup back button
+        backButton.setImage(backImage, for: .normal)
+        backButton.tintColor = playbackButtonsFillColor
+        backButton.addTarget(self, action: #selector(TimeSlider.backAction(_:)), for: .touchUpInside)
+        backButton.showsTouchWhenHighlighted = true
+        addSubview(backButton)
+        
+        // Refresh
+        setNeedsLayout()
```
Comment (philium) [inline Toolkit/ArcGISToolkit/TimeSlider.swift]: While not harmful, I don't think this is needed. The system should request that the view layout its subviews before it is displayed.

File: Toolkit/ArcGISToolkit/TimeSlider.swift (hunk 1 above)
Comment (njarecha) [reply Toolkit/ArcGISToolkit/TimeSlider.swift]: Moved inside `setupUI()`

File: Toolkit/ArcGISToolkit/TimeSlider.swift (hunk 2 above)
Comment (njarecha) [reply Toolkit/ArcGISToolkit/TimeSlider.swift]: fixed

File: Toolkit/ArcGISToolkit/TimeSlider.swift (hunk 3 above)
Comment (njarecha) [reply Toolkit/ArcGISToolkit/TimeSlider.swift]: Removed

**Comment by njarecha** [comment]
@philium Thanks for the review. Addressed all the comments. Please re-review.

**Pull request merged**